### PR TITLE
Feature: Changes for CI/CD dev and staging environment

### PR DESCRIPTION
**Pull request by tabassumsarah**
- This PR contains following
- Configs for dev and staging yaml
- Cloudformation: add some permissions in policy to make staging environment work

### CODE REVIEW

File: main/cicd/cicd-pipeline/.pnpm-debug.log [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (nabi-rezvani) [inline main/cicd/cicd-pipeline/.pnpm-debug.log]: Do we need to commit this?

File: main/cicd/cicd-pipeline/config/infra/cloudformation.yml [hunk 2]
```diff
@@ -228,6 +231,20 @@ Resources:
       PolicyDocument:
         Version: '2012-10-17'
         Statement:
+          - Action:
+              - 'route53:GetChange'
+              - 'route53:ListHostedZones'
+              - 'route53:CreateHostedZone'
```
Comment (nabi-rezvani) [inline main/cicd/cicd-pipeline/config/infra/cloudformation.yml:237]: which step of the pipeline is creating hosted zones?

File: main/config/settings/dev.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
+# Target AWS region, defaults to us-east-1 if not provided.  Note thet some global system components
```
Comment (nabi-rezvani) [inline main/config/settings/dev.yml:1]: should we get rid of these comments?

File: main/cicd/cicd-pipeline/.pnpm-debug.log (hunk 1 above)
Comment (tabassumsarah) [reply main/cicd/cicd-pipeline/.pnpm-debug.log]: you are right. getting rid of it

File: main/cicd/cicd-pipeline/config/infra/cloudformation.yml (hunk 2 above)
Comment (tabassumsarah) [reply main/cicd/cicd-pipeline/config/infra/cloudformation.yml:237]: I think when it deploys backend stack, in one of the step it complains about not having permission for it. I can check by removing it and deploying it.

Comment (tabassumsarah) [reply main/cicd/cicd-pipeline/config/infra/cloudformation.yml:237]: We will test it while creating ci/cd prod, for this PR leaving it as it is.

File: main/config/settings/dev.yml (hunk 3 above)
Comment (tabassumsarah) [reply main/config/settings/dev.yml:1]: Leaving comments for now as they are acting as documentations.

**Review by nabi-rezvani — APPROVED** [review]

File: main/cicd/cicd-pipeline/config/infra/cloudformation.yml [hunk 4]
```diff
@@ -162,6 +160,11 @@ Resources:
           PolicyDocument:
             Version: '2012-10-17'
             Statement:
+              - Action:
+                  - codestar-connections:UseConnection
+                Effect: Allow
+                Resource:
+                  - !Sub 'arn:aws:codestar-connections:ap-southeast-2:311447612024:connection/1f3e5a49-cc86-44fc-b0af-92ee38a477a6'
```
Comment (jkelabora) [inline main/cicd/cicd-pipeline/config/infra/cloudformation.yml]: Use `${AWS::Region}` and `${AWS::AccountId}`?

File: main/config/settings/dev.yml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+# though future features might use it for capabilities such as usage reports and system alerts.
+rootUserEmail: user@example.com
+
+# THIS SETTING IS ONLY APPLICABLE FOR LOCAL DEVELOPMENT
+# A flag indicating if lambda function role's trust policy (assume role policy) should be automatically adjusted to
+# allow assuming the role from local machine (applicable only for local development)
+#localRoleAutoAdjustTrust: true
+# Optional custom domain name; if empty, use the CloudFront domain name
+domainName: dev.wehiresearch.com
+certificateArn: arn:aws:acm:us-east-1:311447612024:certificate/55c12c93-6c30-474c-baba-0297f1d4e9b4
```
Comment (jkelabora) [inline main/config/settings/dev.yml]: Use env vars here as well?

File: main/config/settings/devstg.yml [hunk 6]
```diff
@@ -0,0 +1,126 @@
+# Target AWS region, defaults to us-east-1 if not provided.  Note thet some global system components
+# (e.g. Lambda@Edge) are always deployed via a stack in us-east-1 no matter the setting here.
+# Note: Please delete the hashtag and empty space for settings to be in effect
+awsRegion: ap-southeast-2
```
Comment (jkelabora) [inline main/config/settings/devstg.yml:4]: Use env var here as well?

File: main/config/settings/devstg.yml [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+# though future features might use it for capabilities such as usage reports and system alerts.
+rootUserEmail: user@example.com
+
+# THIS SETTING IS ONLY APPLICABLE FOR LOCAL DEVELOPMENT
+# A flag indicating if lambda function role's trust policy (assume role policy) should be automatically adjusted to
+# allow assuming the role from local machine (applicable only for local development)
+# localRoleAutoAdjustTrust: true
+# Optional custom domain name; if empty, use the CloudFront domain name
+domainName: stgdev.wehiresearch.com
+certificateArn: arn:aws:acm:us-east-1:311447612024:certificate/55c12c93-6c30-474c-baba-0297f1d4e9b4
```
Comment (jkelabora) [inline main/config/settings/devstg.yml]: Use env var here as well?

**Review by jkelabora — APPROVED** [review]

**Pull request merged**